vital/types/point.h

```diff
@@ -0,0 +1,114 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the interface to a simple point.
+ */
+
+#ifndef KWIVER_VITAL_POINT_H_
+#define KWIVER_VITAL_POINT_H_
+
+#include <vital/vital_config.h>
+#include <vital/vital_export.h>
+#include <vital/types/vector.h>
+#include <vital/types/covariance.h>
+
+namespace kwiver {
+namespace vital {
+
+template < unsigned N, typename T >
+class VITAL_EXPORT point
+{
+  typedef std::shared_ptr<covariance_<N, T>> covariance_type;
```

Comment:
Do these types need to be public since they are in the API?

Reply:
Yeah, I think that would be fine

Reply:
Please don't use a `shared_ptr` here. Making the covariance inline, even if it means adding a `bool` flag whether or not it has a meaningful value, is going to be far better performing than the memory indirection plus reference counting of a `shared_ptr`.

Similarly, please carefully consider if you really need to put this type (`point`) in a `shared_ptr`, either, for the same reason.

Reply:
I agree we should not use `shared_ptr` here.  The entire point of this class is to group a location with a covariance and a location so it's not optional, otherwise you would just use `vector_type` directly.  There is also no reason to "share" covariances between points and the covariance itself has a small memory footprint.

Reply:
I am also adding covariance on the geo_point, i will not make that a shared_ptr as well